webhooks-extension/endpoints/webhook.go

```diff
@@ -126,12 +154,153 @@ func (r Resource) createWebhook(request *restful.Request, response *restful.Resp
+		if err != nil {
+			theError := errors.New("bad request information provided, cannot handle deletepipelineruns query (should be set to true or not provided)")
+			logging.Log.Error(theError)
+			RespondError(response, err, http.StatusInternalServerError)
+			return
+		}
+	}
+
+	if namespace == "" {
+		theError := errors.New("no namespace was provided")
+		logging.Log.Error(theError)
+		RespondError(response, theError, http.StatusBadRequest)
+		return
+	}
+
+	logging.Log.Debugf("in deleteWebhook, name: %s, namespace: %s, delete pipeline runs: %s", name, namespace, deletePipelineRuns)
+
+	if name != "" {
+		foundRepoURL := r.findRepoURLFromConfigMap(name, namespace)
+		// This will remove any PipelineRuns too if specified to, hence the need for the repository URL
+		err = r.deleteGitHubWebhookByName(name, namespace, foundRepoURL, toDeletePipelineRuns)
+		if err != nil {
+			if strings.Contains(err.Error(), "could not find webhook with name") {
+				logging.Log.Errorf("webhook (name %s) not found in namespace %s", name, namespace)
+				RespondError(response, err, http.StatusNotFound)
+				return
+			}
+			logging.Log.Errorf("error deleting the webhook (name %s), error: %s", name, err)
+			RespondError(response, err, http.StatusInternalServerError)
+			return
+
+		} else {
+			logging.Log.Infof("Deleted the webhook %s OK, deleting from ConfigMap next", name)
+		}
+
+		err = r.deleteWebhookFromConfigMapByName(name, namespace)
+		if err != nil {
+			logging.Log.Errorf("error deleting the webhook information (name %s) from the ConfigMap, error: %s", name, err)
+			RespondError(response, err, http.StatusInternalServerError)
+			return
+		}
+	} else {
+		logging.Log.Error("no name was provided")
+		RespondError(response, err, http.StatusBadRequest)
+		return
+	}
+	response.WriteHeader(200)
+}
+
+// Find the repository URL for a webhook. We store webhook in a ConfigMap
+func (r Resource) findRepoURLFromConfigMap(webhookName, namespace string) string {
+	foundURL := ""
+	configMapClient := r.K8sClient.CoreV1().ConfigMaps(namespace)
+
+	configMap, err := configMapClient.Get(ConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		logging.Log.Errorf("couldn't find ConfigMap %s in namespace %s", ConfigMapName, namespace)
+		return ""
+	}
+
+	data := configMap.BinaryData["GitHubSource"]
+
+	itsANestedMap := map[string]map[string]string{}
+	err = json.Unmarshal(data, &itsANestedMap)
+	if err != nil {
+		logging.Log.Errorf("invalid data format in the ConfigMap, can't modify it, error: %s", err.Error())
+	}
+
+	foundURL = itsANestedMap[webhookName]["gitrepositoryurl"]
+	return foundURL
+}
+
+// Delete the webhook information from our ConfigMap
+func (r Resource) deleteWebhookFromConfigMapByName(webhookName, namespace string) error {
+	// We don't want any other requests going through here so use a mutex
+	// otherwise we'll end up with operations performed on old configmaps so we'll have invalid data
```

Comment:
I think these comments are leftover from before you moved the lock code. They can probably be moved/removed now?

Reply:
Yep can go, done